Automattic/wp-calypso
# Gutenframe: Move to Calypso the server-side handler of the iframe connections

#### Changes proposed in this Pull Request

First step of #32044.

This PR starts exploring the approach discussed on p7jreA-2gu-p2 in order to share the code that handles the block editor iframe connections in both Simple and JP sites.

To do so, I created a new package `@automattic/wpcom-block-editor` in the Calypso monorepo to include in it all the shared utilities we need for the Gutenberg in WP.com integration. The first file I included on the package contains the server-side handler of the iframe connections (currently maintained in the WP.com codebase).

The package has been added to the `apps` dir since this is a package we have to build but not to publish (similar to `o2-blocks`: https://github.com/Automattic/wp-calypso/pull/32058).

The source code lives on the `src/iframe-bridge-server.js` file. It contains the same logic that the `gutenberg-wpcom-iframe-bridge.js` file currently maintained in WP.com, but using ES2015+ syntax. I used the `server` suffix on the filename with the idea of having a future `iframe-bridge-client.js` file which should define the client-side handler of the iframe connections (currently maintained in Calypso in the `calypsoify-iframe.jsx` file). That way, it'll be more obvious that the logic of both files should be in sync.

#### To follow-up in future PRs:
* Move more files to Calypso (`gutenberg-wpcom-classic-block-tinymce-plugin.js` and `gutenberg-wpcom.js`).
* Make the build available from a CDN and change WP.com to retrieve the files from there (and replicate it on JP).
* Document the deployment process.

#### Testing instructions

* Build the package: `npx lerna run build --scope='@automattic/wpcom-block-editor'`.
* Upload the build file (`/apps/wpcom-block-editor/dist/iframe-bridge-server.js`) to your sandbox and overwrite the content of the `gutenberg-wpcom-iframe-bridge.js` file with the content of the build file.
* Load `https://wordpress.com/block-editor` and select your sandbox site.
* Make sure that all the existing Calypso integrations (media modal, revisions, trash post, go back to all posts, preview, etc) work as expected.
